tests/ti.blob.addontest.js

```diff
@@ -0,0 +1,69 @@
+/*
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2020 by Axway, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+/* eslint-env mocha */
+/* eslint no-unused-expressions: "off" */
+'use strict';
+var should = require('./utilities/assertions');
+
+describe('Titanium.Blob', function () {
+	var win;
+
+	afterEach(function (done) {
+		if (win) {
+			// If `win` is already closed, we're done.
+			let t = setTimeout(function () {
+				if (win) {
+					win = null;
+					done();
+				}
+			}, 3000);
+
+			win.addEventListener('close', function listener () {
+				clearTimeout(t);
+
+				if (win) {
+					win.removeEventListener('close', listener);
+				}
+				win = null;
+				done();
+			});
+			win.close();
```

Comment:
It also makes me think - I know traditionally our APIs are very event-based for async operations, but we might want to consider starting to add optional callback args at the end of methods that fire these async events as a different, more Node-style, means of interacting with them. i.e.
```javascript
win.close(() => done());
```
It's certainly less verbose for simple operations and avoids the need to remove listeners. (Note that Node's EventEmitter does have one-shot listener APIs, but we never really make use of them in our examples so who knows if they actually work as intended on our proxies - `win.once('close', () => {});`)

Reply:
That would be nice! Or even better:
```JS
win.close().then(_ => {
    // Window closed.
});
```